.github/workflows/ci.yml

```
@@ -0,0 +1,33 @@
+name: Continuous Integration
+
+on:
+  push:
+
+env:
+  AWS_REGION: ${{ vars.AWS_REGION }}
+  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+  APP_ID: ${{ secrets.APP_ID }}
+  ENV_ID: ${{ secrets.ENV_ID }}
+  CONFIG_ID: ${{ secrets.CONFIG_ID }}
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Run tests
+        run: docker-compose -f "action-services.yml" run app bash -c "./scripts/test.sh"
+
+  deploy:
+    name: Deploy
+    if: github.ref_name == 'main'
+    needs: test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Deploy lambda
```

I think this will work, and it will save run time. Each job is billed to the minute, rounded up.
```suggestion
      - name: Deploy lambda
      if: github.ref_name == 'main'
```

---

I'd be more concerned about it if it was a private repo. Is it worth doing it for a public repo?

---

Good point. Not for the run time reason, but maybe just for simplicity.